cmake/modules/RootNewMacros.cmake

```diff
@@ -1079,6 +1079,22 @@ function(ROOT_ADD_TEST test)
 endfunction()
 
 #----------------------------------------------------------------------------
+# function ROOT_ADD_GTEST(<testsuite> <name> LIBRARIES)
+#
+function(ROOT_ADD_GTEST test_suite test_name)
+  CMAKE_PARSE_ARGUMENTS(ARG "" "" "LIBRARIES" ${ARGN})
+
+  include_directories(${GTEST_INCLUDE_DIR} ${GMOCK_INCLUDE_DIR})
+
+  # Note we cannot use ROOT_EXECUTABLE because it requires to pass LIBRARIES to link with.
+  # The test suites should chose this in their specific CMakeLists.txt file.
+  ROOT_EXECUTABLE(${test_suite} ${test_name} LIBRARIES ${ARG_LIBRARIES})
```

Comment:
That's not clear, I don't remember why I put this comment there. I think it means to say, we cannot just call ROOT_EXECUTABLE from the outside because we need to link against the gtest libraries or we will have to explicitly specify them outside.